filebeat/harvester/reader/multiline.go

```diff
@@ -113,40 +115,40 @@ func (mlr *Multiline) Next() (Message, error) {
 }
 
 func (mlr *Multiline) readFirst() (Message, error) {
-	for {
-		message, err := mlr.reader.Next()
-		if err != nil {
-			// no lines buffered -> ignore timeout
-			if err == sigMultilineTimeout {
-				continue
-			}
+	//for {
+	message, err := mlr.reader.Next()
+	if err != nil {
+		// no lines buffered -> ignore timeout
+		//if err == sigMultilineTimeout {
+		//	continue
+		//}
 
-			// pass error to caller (next layer) for handling
-			return message, err
-		}
+		// pass error to caller (next layer) for handling
+		return message, err
+	}
 
-		if message.Bytes == 0 {
-			continue
-		}
+	//if message.Bytes == 0 {
+	//	continue
+	//}
 
-		// Start new multiline event
-		mlr.clearBuffer()
-		mlr.resetBuffer(message)
-		mlr.setState((*Multiline).readNext)
-		return mlr.readNext()
-	}
+	// Start new multiline event
+	mlr.clearBuffer()
+	mlr.resetBuffer(message)
+	mlr.setState((*Multiline).readNext)
+	return mlr.readNext()
+	//}
 }
 
 func (mlr *Multiline) readNext() (Message, error) {
 	for {
 		message, err := mlr.reader.Next()
 		if err != nil {
 			// handle multiline timeout signal
-			if err == sigMultilineTimeout {
+			if err == sigMultilineTimeout || err == io.EOF {
```

Comment:
When pushing the buffer on io.EOF it must be very clear, no new line will ever be read from said file. Otherwise we've the chance of pushing a partial event

Does bottom-most log file reader ever return EOF?